docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+# ADR-102: RPC Companion
+
+## Changelog
+
+- 2022-03-27: First draft (@andynog)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+This ADR proposes an architecture of an ***RPC Companion*** solution implemented based on the proposed [ADR-101 Data Companion Pull API](https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md). This solution can run as a sidecar concurrently with the full node, and it is optional.
+
+This ADR provides a reference implementation of a system that can be used to offload queryable data from a CometBFT
+full node to a database and offering a service that exposes the same JSONRPC methods on a endpoint as the regular JSONRPC methods of a CometBFT node endpoint,
+which makes it easier for integrators of RPC clients such as client libraries and applications to switch to this
+***RPC Companion*** with as minimum effort as possible.
+
+This architecture also makes it possible to scale horizontally the querying capacity of a full node by running multiple
+copies of the ***RPC Companion*** server instances that can be behind a scalable load-balancer (e.g. Cloudflare)  and can serve
+the data in a more reliable way.
+
+## Alternative Approaches
+
+Currently, there aren't any alternative solutions that are based on a data companion architecture. It is
```

Comment:
I would rephrase that in a way that makes it clear that ADR 101 is a new feature that will potentially be adopted by users once it is well-established. And that this ADR proposes a first use-case for this feature.

Reply:
reworded, thanks